x-pack/plugins/security_solution/cypress/tasks/alerts.ts

```diff
@@ -80,94 +79,54 @@ import { visit } from './login';
 
 export const addExceptionFromFirstAlert = () => {
   expandFirstAlertActions();
-  cy.get(ADD_EXCEPTION_BTN, { timeout: 10000 }).should('be.visible');
   cy.get(ADD_EXCEPTION_BTN, { timeout: 10000 }).first().click();
   cy.get(LOADING_SPINNER).should('exist');
   cy.get(LOADING_SPINNER).should('not.exist');
 };
 
 export const openAddEndpointExceptionFromFirstAlert = () => {
   expandFirstAlertActions();
-  cy.root()
-    .pipe(($el) => {
-      $el.find(ADD_ENDPOINT_EXCEPTION_BTN).trigger('click');
-      return $el.find(FIELD_INPUT);
-    })
-    .should('be.visible');
+  cy.get(ADD_ENDPOINT_EXCEPTION_BTN).click();
+  cy.get(FIELD_INPUT).should('be.visible');
 };
 
 export const openAddExceptionFromAlertDetails = () => {
   cy.get(EXPAND_ALERT_BTN).first().click({ force: true });
 
-  cy.root()
-    .pipe(($el) => {
-      $el.find(TAKE_ACTION_BTN).trigger('click');
-      return $el.find(TAKE_ACTION_MENU);
-    })
-    .should('be.visible');
+  cy.get(TAKE_ACTION_BTN).click();
+  cy.get(TAKE_ACTION_MENU).should('be.visible');
 
-  cy.root()
-    .pipe(($el) => {
-      $el.find(ADD_EXCEPTION_BTN).trigger('click');
-      return $el.find(ADD_EXCEPTION_BTN);
-    })
-    .should('not.be.visible');
+  cy.get(ADD_EXCEPTION_BTN).click();
+  cy.get(ADD_EXCEPTION_BTN).should('not.be.visible');
 };
 
 export const closeFirstAlert = () => {
   expandFirstAlertActions();
-  cy.get(CLOSE_ALERT_BTN)
-    .pipe(($el) => $el.trigger('click'))
-    .should('not.exist');
+  cy.get(CLOSE_ALERT_BTN).click();
+  cy.get(CLOSE_ALERT_BTN).should('not.exist');
 };
 
 export const closeAlerts = () => {
-  cy.get(TAKE_ACTION_POPOVER_BTN)
-    .first()
-    .pipe(($el) => $el.trigger('click'))
-    .should('be.visible');
-
-  cy.get(CLOSE_SELECTED_ALERTS_BTN)
-    .pipe(($el) => $el.trigger('click'))
-    .should('not.be.visible');
+  cy.get(TAKE_ACTION_POPOVER_BTN).first().click();
+  cy.get(TAKE_ACTION_POPOVER_BTN).should('be.visible');
+  cy.get(CLOSE_SELECTED_ALERTS_BTN).click();
+  cy.get(CLOSE_SELECTED_ALERTS_BTN).should('not.be.visible');
 };
 
 export const expandFirstAlertActions = () => {
   waitForAlerts();
 
-  let count = 0;
-
-  const click = ($el: JQuery<HTMLElement>) => {
-    count++;
-    return $el.trigger('click');
-  };
-
-  /*
-   *
-   * Sometimes it takes some time for UI to attach event listener to
-   * TIMELINE_CONTEXT_MENU_BTN and cypress is too fast to click.
-   * Becuase of this popover does not open when click.
-   * pipe().should() makes sure that pipe function is repeated until should becomes true
-   *
-   * */
-
+  cy.get(TIMELINE_CONTEXT_MENU_BTN).first().filter(':visible').click();
+  // cy.get(TIMELINE_CONTEXT_MENU_BTN).first().should('be.visible').click();
   cy.get(TIMELINE_CONTEXT_MENU_BTN)
     .first()
     .should('be.visible')
-    .pipe(click)
```

Comment:
yes, Cypress has builtin retry-ability, so it shouldn't be needed to add custom retries
https://docs.cypress.io/guides/core-concepts/retry-ability

Reply:
The previous code would try to click, then run the assertions that followed (i.e. the `.should` calls) and then if those failed, it would retry the entire command, including the `click`. Without `pipe`, the `click` will not be retried, even with Cypress's built in retry feature. 

Here is the code comment above. It explains that the click needs to be retried:

> /*
>  *
>  * Sometimes it takes some time for UI to attach event listener to
>  * TIMELINE_CONTEXT_MENU_BTN and cypress is too fast to click.
>  * Becuase of this popover does not open when click.
>  * pipe().should() makes sure that pipe function is repeated until should becomes true
>  *
>  * */

This use case for `pipe` is described in this Cypress blog post: https://www.cypress.io/blog/2019/01/22/when-can-the-test-click/

Perhaps it is not true that the popover's event listener binds late, in which case this `click` retry is not needed. However, if the popover does bind it's click late, then this change might introduce flake instead of reducing it.

Reply:
we can always use `cypress-recurse` if we find the test being flaky

Reply:
I understand your point @oatkiller, but we need to find a balance between wrapping everything into the `cypress-recurse` vs making sure that the actions we take are not having side-effects, if so I think we should be looking for the root cause somewhere else because if the popover randomly disappears it means something is wrong with our UI, not with the tests itself. It means also that our users will be able to face the same issue while they are using our product, which is nothing that we would rather try to avoid than masking it with extra clicks in Cypress